test/resources/auction_examples.js

```diff
@@ -0,0 +1,335 @@
+    prices: [1, 2].map(toETH),
+    owners: [0, 1],
+    tokenIdsForPrice: [0, 1],
+    buyVolumes: [toETH(9), feeSubtracted(toETH(18))]
+  }
+]
+
+/////--------------- Advanced Trade
+
+const advancedTrade = {
+  deposits: [
+    { amount: feeAdded(toETH(20)).add(amountEpsilon), token: 0, user: 0 },
+    { amount: toETH(10), token: 1, user: 1 },
+    { amount: feeAdded(toETH(20)).add(amountEpsilon), token: 0, user: 2 },
+    { amount: toETH(10), token: 1, user: 3 },
+    { amount: feeAdded(toETH(20)).add(amountEpsilon), token: 0, user: 4 },
+    { amount: toETH(10), token: 1, user: 5 },
+  ],
+  orders: [
+    { sellToken: 0, buyToken: 1, sellAmount: feeAdded(toETH(20)).add(amountEpsilon), buyAmount: toETH(10), user: 0 },
+    { sellToken: 1, buyToken: 0, sellAmount: toETH(10), buyAmount: feeSubtracted(toETH(20)).sub(amountEpsilon), user: 1 },
+    { sellToken: 0, buyToken: 1, sellAmount: feeAdded(toETH(20)).add(amountEpsilon), buyAmount: toETH(10), user: 2 },
+    { sellToken: 1, buyToken: 0, sellAmount: toETH(10), buyAmount: feeSubtracted(toETH(20)).sub(amountEpsilon), user: 3 },
+    { sellToken: 0, buyToken: 1, sellAmount: feeAdded(toETH(20)).add(amountEpsilon), buyAmount: toETH(10), user: 4 },
+    { sellToken: 1, buyToken: 0, sellAmount: toETH(10), buyAmount: feeSubtracted(toETH(20)).sub(amountEpsilon), user: 5 },
+  ],
+}
+
+const zero = new BN(0)
+const advancedTradeSolutions = [
+  {
+    name: "Match 1 pair",
+    prices: [1, 2].map(toETH),
+    owners: [0, 1],
+    tokenIdsForPrice: [0, 1],
+    buyVolumes: [toETH(10), feeSubtracted(toETH(20)), zero, zero, zero, zero]
+  },
+  {
+    name: "Match 2 pairs",
+    prices: [1, 2].map(toETH),
+    owners: [0, 1, 2, 3],
+    tokenIdsForPrice: [0, 1],
+    buyVolumes: [toETH(10), feeSubtracted(toETH(20)), toETH(10), feeSubtracted(toETH(20)), zero, zero]
+  },
+  {
+    name: "Match 3 pairs",
+    prices: [1, 2].map(toETH),
+    owners: [0, 1, 2, 3, 4, 5],
+    tokenIdsForPrice: [0, 1],
+    buyVolumes: [toETH(10), feeSubtracted(toETH(20)), toETH(10), feeSubtracted(toETH(20)), toETH(10), feeSubtracted(toETH(20))]
+  }
+]
+
+
+/////--------------- One large (market maker) and one small (market order)
+
+const biggieSmallTrade = {
+  deposits: [
+    { amount: toETH(185), token: 0, user: 0 },
+    { amount: toETH(1000), token: 1, user: 1 },
+  ],
+  orders: [
+    { sellToken: 0, buyToken: 1, sellAmount: toETH(185), buyAmount: toETH(1), user: 0 },
+    { sellToken: 1, buyToken: 0, sellAmount: toETH(1000), buyAmount: toETH(184000), user: 1 },
+  ]
+}
+
+const biggieSmallTradeSolutions = [
+  {
+    name: "Max Fulfillment",
+    // prices: [toETH(1), new BN("184184184184184185000")],
+    prices: [toETH(1), new BN("184184184184184184184")],
+    owners: [0, 1],
+    tokenIdsForPrice: [0, 1],
+    // buyVolumes: [toETH(1), new BN("184000000000000000815")],
+    buyVolumes: [toETH(1), new BN("184000000000000000000")],
+    sellVolumes: [new BN("184368552736921106106"), toETH(1)],
```

Comment:
I would use `toETH(184)` here instead of `184000...0`

Reply:
Also, we calculate sell volumes so we don't need to include them in the solution.